uplink/config.go

```diff
@@ -41,10 +44,11 @@ type RSConfig struct {
 // EncryptionConfig is a configuration struct that keeps details about
 // encrypting segments
 type EncryptionConfig struct {
-	Key       string      `help:"root key for encrypting the data"`
-	BlockSize memory.Size `help:"size (in bytes) of encrypted blocks" default:"1KiB"`
-	DataType  int         `help:"Type of encryption to use for content and metadata (1=AES-GCM, 2=SecretBox)" default:"1"`
-	PathType  int         `help:"Type of encryption to use for paths (0=Unencrypted, 1=AES-GCM, 2=SecretBox)" default:"1"`
+	EncryptionKey string      `help:"the root key for encrypting the data; when set, it overrides the key stored in the file indicated by the key-filepath flag"`
+	KeyFilepath   string      `help:"the path to the file which contains the root key for encrypting the data"`
```

Comment:
I've been thinking how to handle this better. I think the major issue with the changes comes from conflating flags and configuration. Maybe, if we separate the Flags and Config struct. This means that things that do need to reuse the arguments.

So we would have something like:

```
func (flags *EncryptionFlags) Config() (*EncryptionConfig, error) {
```

This would get rid of the uplink depending on io issue. When we eventually get a proper place to store `EncryptionFlags` we can move it outside of `uplink` as well.

But, I'm also trying not to drag this PR on longer, because it has suffered long enough. So a separate PR for doing this change might make sense.

Reply:
The annoying part is that it's part of the `uplink.Config` so we would need to split that as well.

Reply:
Also this would allow later (in a separate PR) to clean up `EncryptionConfig` to be more specific:

```
type EncryptionConfig struct {
	Key        storj.Key
	DataCipher storj.CipherSuite // or what ever the actual type was
	PathCipher storj.CipherSuite // or what ever the actual type was
	BlockSize  memory.Size
}
```

Reply:
_PS: all of these can be done as a separate PRs to get things moving._

Reply:
I'm ok with trying it if the time that I would need for it fits to the current sprint planning.
I'll comment about it in the today's standup.

Reply:
Tacked in https://storjlabs.atlassian.net/browse/V3-1812